### DISCUSSION TITLE
USD limitations with general Maya tools

### DISCUSSION THREAD

**Discussion by raphaeljnwave** [Autodesk/maya-usd#2160] [Q&A]
Hello Maya USD team,

We are trying to adopt USD in Maya for our pipeline. We are happy to see that new features are added to MayaUSD update after updater.

But it seems that the whole integration of USD for the Maya tools is not for tomorrow. For example, we cannot add a new attribute to a USD shape, some tools (e.g. layer view, lightlink view,...) that require maya objects are not aware of the existance of the USD Prims, ...
I guess it is related to USD and Maya communicating through the UFE2.0. 

My question is: is it going to evolve ? Do the maya tools eventually deal with USD prim ? Or are we suppose to develop our own tool to directly communicate through the UFE.

Thank you

**Comment by santosd** [comment]
Hello @raphaeljnwave,
We are regularly adding new capabilities to make Maya aware of USD prims and be able to work with USD in a Maya centric way. We have a few things logged internally to work on as we continue forward, but if there are any specific things that are important to your workflow, please let us know or feel free to log them as issues so that we can triage hem and make sure we have them logged internally.
